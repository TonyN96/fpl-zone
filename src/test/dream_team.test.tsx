import React from "react";
import { render } from "@testing-library/react";
import DreamTeam from "../components/dream_team/dream_team";
import { mockPlayers, mockPositions } from "./fixture-data";
import "@testing-library/jest-dom/extend-expect";

describe("Dream Team Tests", () => {
  const createComponent = (): JSX.Element => {
    return <DreamTeam players={mockPlayers} positions={mockPositions} />;
  };

  it("Renders as expected", () => {
    const { asFragment } = render(createComponent());
    expect(asFragment()).toMatchSnapshot();
  });
});
