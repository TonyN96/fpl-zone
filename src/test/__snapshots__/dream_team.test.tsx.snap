// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dream Team Tests Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container css-1odby6v-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-10 css-1sf73xg-MuiGrid-root"
      data-testid="first-xi-players"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-1yrhruj-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
            data-testid="662"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <img
                alt="kit-img"
                data-testid="player-shirt-image"
                height="55"
                src="/assets/images/kits/7.png"
              />
            </div>
            <div
              class="MuiBox-root css-1gn4x21"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1hrwrfg-MuiTypography-root"
                  data-testid="player-name"
                >
                  TRINER
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item player-text-points css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
                  data-testid="player-score"
                >
                  30
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-1yrhruj-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
            data-testid="828"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <img
                alt="kit-img"
                data-testid="player-shirt-image"
                height="55"
                src="/assets/images/kits/3.png"
              />
            </div>
            <div
              class="MuiBox-root css-1gn4x21"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1hrwrfg-MuiTypography-root"
                  data-testid="player-name"
                >
                  JANKIN
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item player-text-points css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
                  data-testid="player-score"
                >
                  13
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
            data-testid="324"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <img
                alt="kit-img"
                data-testid="player-shirt-image"
                height="55"
                src="/assets/images/kits/6.png"
              />
            </div>
            <div
              class="MuiBox-root css-1gn4x21"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1hrwrfg-MuiTypography-root"
                  data-testid="player-name"
                >
                  SUMMERILL
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item player-text-points css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
                  data-testid="player-score"
                >
                  13
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
            data-testid="929"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <img
                alt="kit-img"
                data-testid="player-shirt-image"
                height="55"
                src="/assets/images/kits/2.png"
              />
            </div>
            <div
              class="MuiBox-root css-1gn4x21"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1hrwrfg-MuiTypography-root"
                  data-testid="player-name"
                >
                  MALLOY
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item player-text-points css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
                  data-testid="player-score"
                >
                  13
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
            data-testid="156"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <img
                alt="kit-img"
                data-testid="player-shirt-image"
                height="55"
                src="/assets/images/kits/14.png"
              />
            </div>
            <div
              class="MuiBox-root css-1gn4x21"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1hrwrfg-MuiTypography-root"
                  data-testid="player-name"
                >
                  TURBEFIELD
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item player-text-points css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
                  data-testid="player-score"
                >
                  9
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-1yrhruj-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
            data-testid="505"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <img
                alt="kit-img"
                data-testid="player-shirt-image"
                height="55"
                src="/assets/images/kits/18.png"
              />
            </div>
            <div
              class="MuiBox-root css-1gn4x21"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1hrwrfg-MuiTypography-root"
                  data-testid="player-name"
                >
                  MULLER
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item player-text-points css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
                  data-testid="player-score"
                >
                  24
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
            data-testid="938"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <img
                alt="kit-img"
                data-testid="player-shirt-image"
                height="55"
                src="/assets/images/kits/2.png"
              />
            </div>
            <div
              class="MuiBox-root css-1gn4x21"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1hrwrfg-MuiTypography-root"
                  data-testid="player-name"
                >
                  PERRINGTON
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item player-text-points css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
                  data-testid="player-score"
                >
                  16
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
            data-testid="137"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <img
                alt="kit-img"
                data-testid="player-shirt-image"
                height="55"
                src="/assets/images/kits/19.png"
              />
            </div>
            <div
              class="MuiBox-root css-1gn4x21"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1hrwrfg-MuiTypography-root"
                  data-testid="player-name"
                >
                  TINGLEY
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item player-text-points css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
                  data-testid="player-score"
                >
                  12
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-1yrhruj-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
            data-testid="466"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <img
                alt="kit-img"
                data-testid="player-shirt-image"
                height="55"
                src="/assets/images/kits/19.png"
              />
            </div>
            <div
              class="MuiBox-root css-1gn4x21"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1hrwrfg-MuiTypography-root"
                  data-testid="player-name"
                >
                  WORKES
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item player-text-points css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
                  data-testid="player-score"
                >
                  22
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-2 css-1djelbh-MuiGrid-root"
      data-testid="bench-players"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-fkexd4-MuiTypography-root"
        >
          BENCH
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
          data-testid="9"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="kit-img"
              data-testid="player-shirt-image"
              height="55"
              src="/assets/images/kits/7.png"
            />
          </div>
          <div
            class="MuiBox-root css-1gn4x21"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1hrwrfg-MuiTypography-root"
                data-testid="player-name"
              >
                BAULDREY
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item player-text-points css-13i4rnv-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
                data-testid="player-score"
              >
                23
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
          data-testid="286"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="kit-img"
              data-testid="player-shirt-image"
              height="55"
              src="/assets/images/kits/5.png"
            />
          </div>
          <div
            class="MuiBox-root css-1gn4x21"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1hrwrfg-MuiTypography-root"
                data-testid="player-name"
              >
                MACMENEMY
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item player-text-points css-13i4rnv-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
                data-testid="player-score"
              >
                7
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
          data-testid="428"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="kit-img"
              data-testid="player-shirt-image"
              height="55"
              src="/assets/images/kits/13.png"
            />
          </div>
          <div
            class="MuiBox-root css-1gn4x21"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1hrwrfg-MuiTypography-root"
                data-testid="player-name"
              >
                MACEK
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item player-text-points css-13i4rnv-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
                data-testid="player-score"
              >
                7
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-t0zib5-MuiGrid-root"
          data-testid="756"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="kit-img"
              data-testid="player-shirt-image"
              height="55"
              src="/assets/images/kits/14.png"
            />
          </div>
          <div
            class="MuiBox-root css-1gn4x21"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1hrwrfg-MuiTypography-root"
                data-testid="player-name"
              >
                MIETTINEN
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item player-text-points css-13i4rnv-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
                data-testid="player-score"
              >
                6
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
